internal/clusterconfig/clusterconfig.go

```diff
@@ -168,34 +164,24 @@ func (r *UpgradeClusterConfigGather) fetchProxy(ctx context.Context, manifestsDi
 	return utils.MarshalToFile(p, filePath)
 }
 
-func (r *UpgradeClusterConfigGather) fetchMachineConfigs(ctx context.Context, manifestsDir string) error {
-	for _, machineConfigName := range machineConfigNames {
-		r.Log.Info("Fetching MachineConfig", "name", machineConfigName)
+func (r *UpgradeClusterConfigGather) fetchSSHPublicKey(ctx context.Context,
+	seedReconfiguration *seedreconfig.SeedReconfiguration) error {
 
-		machineConfig := mcv1.MachineConfig{}
-		if err := r.Client.Get(ctx, types.NamespacedName{Name: machineConfigName}, &machineConfig); err != nil {
-			return err
-		}
-
-		mc := mcv1.MachineConfig{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:   machineConfig.Name,
-				Labels: machineConfig.Labels,
-			},
-			Spec: machineConfig.Spec,
-		}
-		machineConfigTypeMeta, err := r.typeMetaForObject(&machineConfig)
-		if err != nil {
-			return err
-		}
-		mc.TypeMeta = *machineConfigTypeMeta
+	machineConfig := mcv1.MachineConfig{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: masterSSHMachineConfig}, &machineConfig); err != nil {
+		return err
+	}
 
-		filePath := filepath.Join(manifestsDir, machineConfig.Name+".json")
-		r.Log.Info("Writing MachineConfig to file", "path", filePath)
-		if err := utils.MarshalToFile(mc, filePath); err != nil {
-			return err
+	config, _, err := ignition.Parse(machineConfig.Spec.Config.Raw)
+	if err != nil {
+		return err
+	}
+	for _, user := range config.Passwd.Users {
```

Comment:
maybe we should take all the list as map[user]=keyList ?
or just leave machine config files as is and only provide key for early connection before cluster is up?

Reply:
No need for list, the openshift installer is the one generating these MC and the installer API accepts a single ssh key